conan/cache/cache.py

```diff
@@ -51,35 +43,30 @@ def base_folder(self):
         return self._base_folder
 
     @staticmethod
-    def _short_hash_path(hash):
-        """:param hash: Unicode text to reduce"""
-        hash = hash.encode("utf-8")
+    def _short_hash_path(h):
+        """:param h: Unicode text to reduce"""
+        h = h.encode("utf-8")
         md = hashlib.sha256()
-        md.update(hash)
+        md.update(h)
         sha_bytes = md.hexdigest()
         # len based on: https://github.com/conan-io/conan/pull/9595#issuecomment-918976451
         return sha_bytes[0:16]
 
     @staticmethod
     def _get_tmp_path():
-        hash = DataCache._short_hash_path(str(uuid.uuid4()))
-        return os.path.join("tmp", hash)
+        h = DataCache._short_hash_path(str(uuid.uuid4()))
+        return os.path.join("tmp", h)
 
     @staticmethod
     def _get_path(ref: ConanReference):
         value = ref.full_reference
         return DataCache._short_hash_path(value)
 
-    def create_tmp_reference_layout(self, ref: ConanReference):
-        assert not ref.rrev, "Recipe revision should be unknown"
-        # even if they are temporal, we have to assign unique revisions to temporal references
-        # until we know the calculated revision so that we don't store multiple references
-        # with same rrev or prev
-        temp_rrev = f"{RREV_UNKNOWN}-{str(uuid.uuid4())}"
-        ref = ConanReference(ref.name, ref.version, ref.user, ref.channel, temp_rrev,
-                             ref.pkgid, ref.prev)
+    def create_export_recipe_layout(self, ref: ConanReference):
```

Comment:
Tried to add the ``export`` word here, to explicit the use case, this only happens when exporting a recipe to the cache. As the implementation shows, this has not entry in the DB now